src/v/cluster/topics_frontend.cc

```diff
@@ -216,6 +216,51 @@ cluster::errc map_errc(std::error_code ec) {
     return errc::replication_error;
 }
 
+namespace {
+bool has_enterprise(
+  const metadata_cache& metadata, const topic_properties_update& update) {
+    using op = incremental_update_operation;
+    using si_mode = model::shadow_indexing_mode;
+
+    constexpr auto is_set_true = [](const auto& v) {
+        return v.op == op::set && v.value == true;
+    };
+
+    constexpr auto maybe_remove_flag = [](auto& mode, auto upd, auto flag) {
+        if (upd.op == op::remove || (upd.op == op::set && upd.value == false)) {
+            mode = add_shadow_indexing_flag(mode, flag);
+        }
+    };
+
+    const auto& update_props = update.properties;
+
+    if (
+      is_set_true(update_props.remote_read)
+      || is_set_true(update_props.remote_write)
+      || is_set_true(update_props.remote_delete)) {
+        return true;
+    }
+
+    // Build what the final mode would be.
+    auto mode = metadata.get_default_shadow_indexing_mode();
+    auto meta = metadata.get_topic_metadata_ref(update.tp_ns);
+    if (meta.has_value()) {
+        mode
+          = meta->get().get_configuration().properties.shadow_indexing.value_or(
+            mode);
+    };
+
+    maybe_remove_flag(mode, update_props.remote_read, si_mode::drop_fetch);
+    maybe_remove_flag(mode, update_props.remote_write, si_mode::drop_archival);
+
+    if (mode != si_mode::disabled) {
```

Comment:
We still want to allow them to alter non-enterprise topic configs (e.g. `retention.ms`) for their already-tiered storage-enabled topics.
```suggestion
    if (mode != si_mode::disabled || mode == current_mode) {
```

I think we should also drop 237-242 and have a similar check for `remote_delete` as well, where we check if it becomes enabled && it was not enabled.

There's a subtle but important behaviour difference here based on whether the tiered storage topic properties are explicitly specified or not. The case where this matters that I am thinking about is when customers manage their topic properties with terraform. As far as know, the terraform providers issue AlterTopicConfigs updates with all topic configs specified. Let's say they have an existing topic with TS enabled, their license expired yesterday, and they want to update `retention.ms`. With the existing code, they wouldn't be able to do that because terraform's AlterTopicsRequest would specify the enabled tiered storage topic properties explicitly as enabled and this check would fail - even though that is the existing state of the topic property and they are just trying to change a non-TS topic property. So I think we want to drop 237-242 and change all checks to be of the style `becomes_enabled(X) && X != old_X`.